sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -240,6 +260,85 @@ string_class ProgramManager::getProgramBuildLog(const RT::PiProgram &Program) {
   return Log;
 }
 
+static bool loadDeviceLib(const RT::PiContext &Context, const char *Name,
+                          RT::PiProgram &Prog) {
+  std::string LibSyclDir = OSUtil::getCurrentDSODir();
+  std::ifstream File(LibSyclDir + OSUtil::DirSep + Name,
+                     std::ifstream::in | std::ifstream::binary);
+  if (!File.good()) {
+    return false;
+  }
+
+  File.seekg(0, std::ios::end);
+  size_t FileSize = File.tellg();
+  File.seekg(0, std::ios::beg);
+  std::vector<char> FileContent(FileSize);
+  File.read(&FileContent[0], FileSize);
+  File.close();
+
+  Prog =
+      createSpirvProgram(Context, (unsigned char *)&FileContent[0], FileSize);
+  return Prog != nullptr;
+}
+
+static std::string getDeviceExtensions(const RT::PiDevice &Dev) {
```

Comment:
`detail::get_device_info<...>` methods accept `RT::PiDevice`, not `detail::device_impl`. And if you want to avoid using a vector, you can call `detail::get_device_info<string_class, info::device::extensions>::get(RT::PiDevice)` instead, which returns a single string.

Reply:
Oh, I wasn't aware about this function. Fixed.